client cleanup

SuperchupuDev opened the pull request:
- [x] reduce client type/variable duplication
- [x] add loadDirectory/loadFile
- [x] update handlers
- [x] update subcommand handler
- [x] use subcommand builders
- [x] add deploy support

SuperchupuDev committed:
some wip implementation

SuperchupuDev committed:
more things

SuperchupuDev committed:
Merge branch 'main' into refactor/client-cleanup

SuperchupuDev committed:
move guild checker to ready event

SuperchupuDev committed:
improve error handling in ai

SuperchupuDev committed:
better command loading

SuperchupuDev committed:
.con is no longer needed

SuperchupuDev committed:
no need for this

SuperchupuDev committed:
better subcommand loading

the pain is unbearable

SuperchupuDev committed:
count base subcommands in the log too

SuperchupuDev committed:
better component loading

SuperchupuDev committed:
no more arrow function

SuperchupuDev committed:
ComponentHandler is now cooler

SuperchupuDev committed:
better event loading + reloading all now works

SuperchupuDev committed:
messages arent even emitted in dms looool

SuperchupuDev committed:
source maps

Commit:
more useful ready log

Nyameliaaaa commented on src/classes/Client.ts:
```diff
@@ -23,23 +16,24 @@ import TagHandler from './handlers/TagHandler';
+        'https://discord.com/api/oauth2/authorize?client_id=831932409943425064&scope=bot+applications.commands&permissions=474527689975';
+    version = process.env.npm_package_version;
+    devs = ['707675871355600967', '478823932913516544', '348591272476540928'];
     generating: boolean;
```
this is useless i think
![image](https://user-images.githubusercontent.com/32780515/155684816-8f934d5f-67d9-4741-92f0-dddeb7567038.png)

SuperchupuDev replied:
@xiboon you are the one who added this what do you think

hkubus replied:
im pretty sure this was moved to client.ai.generating

Nyameliaaaa replied:
yep

its isGenerating but ok 👍

SuperchupuDev committed:
Merge branch 'main' into refactor/client-cleanup

Nyameliaaaa committed:
Remove useless client variable

Nyameliaaaa committed:
xdd

SuperchupuDev committed:
separate subcommand builders into its own files

SuperchupuDev committed:
add start log

i know we dont want to update text commands but before it loaded syncronously because it was evaluated on class load, which made it load before the start log. ig this makes the client load faster

Nyameliaaaa committed:
Don't use exports to pass init time

Nyameliaaaa committed:
Remove extra client.conn mutation

Nyameliaaaa committed:
Make readyAt time more accurate

SuperchupuDev committed:
remove duplicated `loadCommands` call

SuperchupuDev committed:
change optional log args type to `unknown`

SuperchupuDev committed:
Merge branch 'main' into refactor/client-cleanup

Nyameliaaaa reviewed (approved):
👍

hkubus reviewed (approved):
Mom told me to review this
